### PR TITLE
Multi-architecture images using buildx

### LINKED ISSUE #1267: support for DOCKER_BUILDKIT=1 environment variable
# Description

I would like to take advantage in my build of DOCKER BUILDKIT. Not only it runs faster, but has significant improvements like FROM ... AS that makes the images smaller.

I checked and it does not look like it is currently possible with the plugin.

I would like the plugin to respect the environment variable DOCKER_BUILDKIT, or have any other way of passing that information to the build engine.

### LINKED ISSUE #1502: Support / documentation for building multi-arch images (e.g. amd64, arm64)
### Description

Last versions of docker support multi-arch images (see https://docs.docker.com/engine/reference/commandline/buildx_build/).
For us, multi-arch images become very relevant as developers more and more start to use Apple Silicon (e.g. M1) machines due to their high-performance capabilities.

I was not able to find a way to build the images using the plugin.

Would be good to either add support or if it's possible already now, document it in official docs.

### Notes

Tried the following configuration:
```xml
<plugin>
        <groupId>io.fabric8</groupId>
        <artifactId>docker-maven-plugin</artifactId>
        <extensions>true</extensions>
        <configuration>
          <images>
            <image>
              <name>${docker.push.registry}/myservice:${project.version}</name>
              <build>
                <buildOptions>
                  <platform>linux/arm64,linux/amd64</platform>
                </buildOptions>
                <dockerFile>${project.basedir}/src/main/docker/Dockerfile</dockerFile>
                <assembly>
                  <inline>
...

```
The error that I've got is:

> DOCKER> Unable to build image [docker-repo.crxmarkets.tech/payables:22.3.0-SNAPSHOT] : {"message":"\"arm64,linux\" is > an invalid component of \"linux/arm64,linux/amd64\": platform specifier component must match \"^[A-Za-z0-9_-]+$\": invalid argument"} (Bad Request: 400) [{"message":"\"arm64,linux\" is an invalid component of \"linux/arm64,linux/amd64\": platform  specifier component must match \"^[A-Za-z0-9_-]+$\": invalid argument"} (Bad Request: 400)

The builds using `docker buildx build` or `docker build` with BuildKit enabled, succeeded:
`DOCKER_BUILDKIT=1 docker build  --platform=linux/amd64,linux/arm64 -t myservice .`

```
docker buildx ls
NAME/NODE        DRIVER/ENDPOINT      STATUS  PLATFORMS
buildxbuilder *  docker-container             
  buildxbuilder0 tcp://localhost:2375 running linux/amd64*, linux/arm64*, linux/riscv64, linux/ppc64le, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6, linux/s390x
default          docker                       
  default        default              running linux/amd64, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
```

### Environment

* docker-maven-plugin version : 0.37.0
* Maven version (`mvn -v`) : 3.6.3, Java version: 11.0.11, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-amd64
* Docker version : 20.10.10, build b485636

### PR DESCRIPTION
**Pull request by chonton**
Closes #1502 
Closes #1267 

Not perfect, but sufficient to start iterating on.

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@chonton : Thanks a lot for this :+1: . Since this has been requested multiple times. Do you think we should also add a dedicated quickstart for this feature? Or maybe update an existing quickstart with a new profile?

**Comment by chonton** [comment]
Quickstart added in #1548

File: src/main/asciidoc/inc/build/_configuration.adoc [hunk 1]
```diff
@@ -25,6 +25,9 @@ described in <<build-buildargs,Build Args>>
 https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#build-image-from-a-dockerfile[Docker Remote API] and are restricted to simple options
 (e.g.: memory, shmsize). If you use the respective configuration options for build options natively supported by the build configuration (i.e. `squash`, `noCache`, `cleanup=remove` for buildoption `forcerm=1` and `args` for build args) then these will override any corresponding options given here. The key-value syntax is the same as when defining environment variables or labels as described in <<misc-env,Setting Environment Variables and Labels>>.
 
+| *buildX*
```
Comment (rohanKanojia) [inline src/main/asciidoc/inc/build/_configuration.adoc:28]: Umm, I have minor doubt. In BuildImageConfiguration I see flag name is `buildx`. Is this a typo?

Comment (chonton) [reply src/main/asciidoc/inc/build/_configuration.adoc]: Good catch.  Updated.

File: src/main/java/io/fabric8/maven/docker/service/ServiceHub.java [hunk 2]
```diff
@@ -159,21 +162,22 @@ public WaitService getWaitService() {
...
-    public ArchiveService getArchiveService() {
-        return archiveService;
+    public BuildXService getBuildXService() {
+        checkDockerAccessInitialization();
+        return buildXService;
     }
 
     /**
-     * Get a service for executing goals on other Maven mojos
+     * Serivce for creating archives
```
Comment (rohanKanojia) [inline src/main/java/io/fabric8/maven/docker/service/ServiceHub.java:175]:
```suggestion
     * Service for creating archives
```

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**